### PR TITLE
handle TypeError when get_mount_info function get None

### PR DESCRIPTION
**Pull request by xiaoge1001**
## Proposed Commit Message
handle TypeError when get_mount_info function get None

Fixes GH-4321

### CODE REVIEW

File: cloudinit/util.py [hunk 1]
```diff
@@ -2782,7 +2782,14 @@ def get_mount_info(path, log=LOG, get_mnt_opts=False):
 
 
 def has_mount_opt(path, opt: str) -> bool:
-    *_, mnt_opts = get_mount_info(path, get_mnt_opts=True)
+    try:
+        *_, mnt_opts = get_mount_info(path, get_mnt_opts=True)
+    except TypeError:
+        LOG.warning(
+            "Get Nothing mount info from %s",
```
Comment (igalic) [inline cloudinit/util.py:2789]: this message reads really confusing

File: cloudinit/util.py [hunk 2]
```diff
@@ -2782,7 +2782,14 @@ def get_mount_info(path, log=LOG, get_mnt_opts=False):
 
 
 def has_mount_opt(path, opt: str) -> bool:
-    *_, mnt_opts = get_mount_info(path, get_mnt_opts=True)
+    try:
+        *_, mnt_opts = get_mount_info(path, get_mnt_opts=True)
```
Comment (igalic) [inline cloudinit/util.py:2786]: why would this function return `None` to begin with?

Comment (xiaoge1001) [reply cloudinit/util.py:2786]: > why would this function return `None` to begin with?

The reason still needs to be further identified. However, when we look at the implementation code of this function, there is some cases of returning None.

Comment (igalic) [reply cloudinit/util.py:2786]: of all of those, only the first and the last would leave no trace in the log

the first one would mean your mountinfo file is in the wrong, or a new unknown, different, format.

the last one would mean no match was found, including no match for `/`!
I'm not sure there are scenarios where that makes sense (at least as a system to run cloud-init on)

File: cloudinit/util.py (hunk 1 above)
Comment (igalic) [reply cloudinit/util.py:2789]: I think this should say: `"Found no mount info for %s",`

**Comment by xiaoge1001** [comment]
> I don't think that the current approach is desired to resolve the issue.

Yes, we may need to fundamentally address why function get_mount_info() returns None.

**Pull request closed without merging**